backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/DiffComputer.java

```diff
@@ -225,20 +225,16 @@ public static AutogenS3DatasetDiff computeS3DatasetDiff(
                     a,
                     b,
                     AutogenS3DatasetBlob::getComponents,
-                    x -> x.getPath().getPath(),
+                    x -> x.getPath().getPath() + x.getS3VersionId(),
```

Comment:
Be careful about just concatenating strings, as they can randomly match with another concatenation. Using some unlikely chars like `[s1, s2]` makes it less likely.

Reply:
will do

Reply:
agreed on the other comment of not requiring versionid for generating the map key. So this comment in no longer valid